### PR TITLE
[SPARK-31886][WEBUI] Fix the wrong coloring of nodes in DAG-viz

### PR DESCRIPTION
**Pull request by sarutak**
### What changes were proposed in this pull request?

This PR fixes a wrong coloring issue in the DAG-viz.
In the Job Page and Stage Page, nodes which are associated with "barrier mode" in the DAG-viz will be colored pale green.
But, with some type of jobs, nodes which are not associated with the mode will also colored.
You can reproduce with the following operation.
```
sc.parallelize(1 to 10).barrier.mapPartitions(identity).repartition(1).collect() 
```
wrong-coloring

In the screen shot above, `repartition` in `Stage 1` is not associated with barrier mode so the corresponding node should not be colored pale green.

The cause of this issue is the logic which chooses HTML elements to be colored is wrong.
The logic chooses such elements based on whether each element is associated with a style class (`clusterId` in the code). 
But when an operation crosses over shuffle (like `repartition` above), a `clusterId` can be duplicated and non-barrier mode node is also associated with the same `clusterId`.

### Why are the changes needed?

This is a bug.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Newly added test case with the following command.
```
build/sbt -Dtest.default.exclude.tags= -Dspark.test.webdriver.chrome.driver=/path/to/chromedriver "testOnly org.apache.spark.ui.ChromeUISeleniumSuite -- -z SPARK-31886"
```

### CODE REVIEW

File: core/src/test/scala/org/apache/spark/ui/RealBrowserUISeleniumSuite.scala [hunk 1]
```diff
@@ -78,6 +78,35 @@ abstract class RealBrowserUISeleniumSuite(val driverProp: String)
     }
   }
 
+
```
Comment (DylanGuedes) [inline core/src/test/scala/org/apache/spark/ui/RealBrowserUISeleniumSuite.scala:81]: nit: extra line

Comment (sarutak) [reply core/src/test/scala/org/apache/spark/ui/RealBrowserUISeleniumSuite.scala:81]: Ah, Thanks.

**Comment by HyukjinKwon** [comment]
cc @gengliangwang FYI

**Review by gengliangwang — APPROVED** [review]
LGTM, thanks for the fix!

**Comment by gengliangwang** [comment]
Thanks, merging to master/3.0

**Pull request closed without merging**